curvefs/src/metaserver/storage/converter.cpp

```diff
@@ -48,6 +50,48 @@ static bool CompareType(const std::string& str, KEY_TYPE keyType) {
     return StringToUl(str, &n) && n == keyType;
 }
 
+NameGenerator::NameGenerator(uint32_t partitionId)
+    : tableName4Inode_(Format(kTypeInode, partitionId)),
+      tableName4S3ChunkInfo_(Format(kTypeS3ChunkInfo, partitionId)),
+      tableName4Dentry_(Format(kTypeDentry, partitionId)),
+      tableName4VolumeExtent_(Format(kTypeVolumeExtent, partitionId)),
+      tableName4InodeAuxInfo_(Format(kTypeInodeAuxInfo, partitionId)) {}
+
+std::string NameGenerator::GetInodeTableName() const {
+    return tableName4Inode_;
+}
+
+std::string NameGenerator::GetS3ChunkInfoTableName() const {
+    return tableName4S3ChunkInfo_;
+}
+
+std::string NameGenerator::GetDentryTableName() const {
+    return tableName4Dentry_;
+}
+
+std::string NameGenerator::GetVolumnExtentTableName() const {
+    return tableName4VolumeExtent_;
+}
+
+std::string NameGenerator::GetInodeAuxInfoTableName() const {
+    return tableName4InodeAuxInfo_;
+}
+
+size_t NameGenerator::GetFixedLength() {
+    size_t length = sizeof(kTypeInode) + sizeof(uint32_t) + strlen(kDelimiter);
+    LOG(INFO) << "Tablename fixed length is " << length;
+    return length;
+}
+
+std::string NameGenerator::Format(KEY_TYPE type, uint32_t partitionId) {
+    char buffer[sizeof(uint32_t)];
+    std::memcpy(buffer, reinterpret_cast<char*>(&partitionId),
+                sizeof(uint32_t));
+    std::ostringstream oss;
```

Comment:
What are the benefits of using oss instead of direct using string

Reply:
> benefits

you can use string streams to convert anything that implements operator `<<` to a string.